plugins/modules/xenserver_guest.py

```diff
@@ -12,34 +12,34 @@
-    values V(none) and V(dhcp) have same effect. More info here:
-    U(https://www.citrix.com/community/citrix-developer/citrix-hypervisor-developer/citrix-hypervisor-developing-products/citrix-hypervisor-staticip.html).'
-  - 'On platforms without official support for network configuration inside a guest OS, network parameters will be written to xenstore
-    C(vm-data/networks/<vif_device>) key. Parameters can be inspected by using C(xenstore ls) and C(xenstore read) tools on \*nix guests or through WMI
-    interface on Windows guests.
-    They can also be found in VM facts C(instance.xenstore_data) key as returned by the module. It is up to the user to implement a boot time
-    scripts or custom agent that will read the parameters from xenstore and configure network with given parameters. Take note that for xenstore
-    data to become available inside a guest, a VM restart is needed hence module will require VM restart if any parameter is changed. This is
-    a limitation of XenAPI and xenstore. Considering these limitations, network configuration through xenstore is most useful for bootstrapping
+  - 'To acquire XenAPI Python library, just run C(pip install XenAPI) on your Ansible Control Node. The library can also be
+    found inside Citrix Hypervisor/XenServer SDK (downloadable from Citrix website). Copy the C(XenAPI.py) file from the SDK
+    to your Python site-packages on your Ansible Control Node to use it. Latest version of the library can also be acquired
+    from GitHub: U(https://raw.githubusercontent.com/xapi-project/xen-api/master/scripts/examples/python/XenAPI/XenAPI.py).'
+  - If no scheme is specified in O(hostname), module defaults to C(http://) because C(https://) is problematic in most setups.
+    Make sure you are accessing XenServer host in trusted environment or use C(https://) scheme explicitly.
+  - To use C(https://) scheme for O(hostname) you have to either import host certificate to your OS certificate store or use
+    O(validate_certs=false) which requires XenAPI library from XenServer 7.2 SDK or newer and Python 2.7.9 or newer.
+  - 'Network configuration inside a guest OS, by using parameters O(networks[].type), O(networks[].ip), O(networks[].gateway)
+    and so on, is supported on XenServer 7.0 or newer for Windows guests by using official XenServer Guest agent support for
+    network configuration. The module tries to detect if such support is available and utilize it, else it uses a custom method
+    of configuration using xenstore. Since XenServer Guest agent only support None and Static types of network configuration,
+    where None means DHCP configured interface, O(networks[].type) and O(networks[].type6) values V(none) and V(dhcp) have
+    same effect. More info here: U(https://www.citrix.com/community/citrix-developer/citrix-hypervisor-developer/citrix-hypervisor-developing-products/citrix-hypervisor-staticip.html).'
```

Comment:
```suggestion
    same effect. More info here: 
    U(https://www.citrix.com/community/citrix-developer/citrix-hypervisor-developer/citrix-hypervisor-developing-products/citrix-hypervisor-staticip.html).'
```

Reply:
This URL is unfortunately dead. Thanks Citrix! We might as well change it to some working one from the Wayback Machine. This one was the original:

`https://web.archive.org/web/20180218110151/https://xenserver.org/blog/entry/set-windows-guest-vm-static-ip-address-in-xenserver.html`

And its still shorter than the current one.

Is usage of URL shorteners like TinyURL advised in Ansible documentation? We could also use that to shorten the URL even more.

Reply:
I would avoid use of URL shorteners. You can always use `L(..., ...)` to add a nicer/shorter title to a link if you think it is too long for `U(...)`.

Reply:
But the problem here is the line length in the code, not in displayed documentation.

Reply:
Usually linters are OK with a line without spaces being too long. If not, we'll add an ignore.txt entry.

Reply:
`ansible-test sanity --test pep8` was failing (see ansibullbot comment above). I assume that is the reason @russoz wanted this change (i.e. move the long URL to the next line)... which is fine for me.

Reply:
@russoz how about using the archive.org URL here?

Reply:
@bvitnik I am actually crafting a tool that makes that for me (pip install andebox if interested, the yaml-doc action)

Reply:
> @russoz how about using the archive.org URL here?

done